pkg/controller/podgc/gc_controller.go

```diff
@@ -238,7 +240,13 @@ func (gcc *PodGCController) gcOrphaned(ctx context.Context, pods []*v1.Pod, node
 			continue
 		}
 		klog.V(2).InfoS("Found orphaned Pod assigned to the Node, deleting.", "pod", klog.KObj(pod), "node", pod.Spec.NodeName)
-		if err := gcc.deletePod(ctx, pod.Namespace, pod.Name); err != nil {
+		condition := &v1.PodCondition{
```

Comment:
@liggitt what do you think about optimizing here and only adding the condition if the pod has any finalizer?

Reply:
optimizations are good if they make sense. I don't know enough about the observers of this condition to know if they would react differently to `orphaned pod → added condition → delete-with-0-grace-period` than a simple `orphaned pod → delete-with-0-grace-period` sequence.

Reply:
Seems it requires some investigation to understand the balance between optimization and code complication to avoid a premature optimization. Maybe investigating the optimization should be another graduation crateria for Beta? @alculquicondor  @liggitt ?

Reply:
An observer could benefit from that extra "added condition" notification, although it's not guaranteed (their controller could be down, for example). In any case, it makes sense to keep it in all cases to stay consistent.

Reply:
Users/controllers can't depend on a separate notification, since the (client-side) watch infrastructure may combine changes, or they may get disconnected and need to relist.